### PR TITLE
[Notifier] Notifier add test for adding transport factories to framework extension

### PR DESCRIPTION
**Pull request by jschaedl**
| Q             | A
| ------------- | ---
| Branch?       | 5.x
| Bug fix?      | no
| New feature?  | no 
| Deprecations? | no 
| Tickets       |  
| License       | MIT
| Doc PR        |  

Follow up of https://github.com/symfony/symfony/pull/40843 and https://github.com/symfony/symfony/pull/40844

Needs to be rebased after both were merged.

See changes: https://github.com/symfony/symfony/pull/40845/commits/ddf61c2daeffece2c19dba9bb05664030190af40

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
Can you please merge this with #40844?

**Comment by OskarStark** [comment]
Oh sorry @nicolas-grekas I already merged #40844 😢 

@jschaedl can you please rebase? Thanks

**Review by nicolas-grekas — APPROVED** [review]

**Comment by OskarStark** [comment]
Can you please check Travis?

File: src/Symfony/Bundle/FrameworkBundle/composer.json [hunk 1]
```diff
@@ -51,6 +51,36 @@
...
+        "symfony/free-mobile-notifier": "^5.3",
+        "symfony/gatewayapi-notifier": "^5.3",
+        "symfony/gitter-notifier": "^5.3",
+        "symfony/google-chat-notifier": "^5.3",
+        "symfony/infobip-notifier": "^5.3",
+        "symfony/iqsms-notifier": "^5.3",
+        "symfony/light-sms-notifier": "^5.3",
+        "symfony/linked-in-notifier": "^5.3",
+        "symfony/mattermost-notifier": "^5.3",
+        "symfony/microsoft-teams-notifier": "^5.3",
```
Comment (jschaedl) [inline src/Symfony/Bundle/FrameworkBundle/composer.json:71]: As soon as the repository `symfony/message-bird-notifier` is available: 
```suggestion
        "symfony/message-bird-notifier": "^5.3",
        "symfony/microsoft-teams-notifier": "^5.3",
```

File: src/Symfony/Bundle/FrameworkBundle/composer.json [hunk 2]
```diff
@@ -51,6 +51,36 @@
...
+        "symfony/mobyt-notifier": "^5.3",
+        "symfony/nexmo-notifier": "^5.3",
+        "symfony/octopush-notifier": "^5.3",
+        "symfony/ovh-cloud-notifier": "^5.3",
+        "symfony/rocket-chat-notifier": "^5.3",
+        "symfony/sendinblue-notifier": "^5.3",
+        "symfony/sinch-notifier": "^5.3",
+        "symfony/slack-notifier": "^5.3",
+        "symfony/smsapi-notifier": "^5.3",
+        "symfony/spot-hit-notifier": "^5.3",
```
Comment (jschaedl) [inline src/Symfony/Bundle/FrameworkBundle/composer.json:82]: As soon as the repository `symfony/sms-biuras-notifier` is available:

```suggestion
        "symfony/sms-biuras-notifier": "^5.3",
        "symfony/spot-hit-notifier": "^5.3",
```

File: src/Symfony/Bundle/FrameworkBundle/Tests/DependencyInjection/FrameworkExtensionTest.php [hunk 3]
```diff
@@ -1873,6 +1874,16 @@ public function testNotifierWithoutTransports()
         $this->assertFalse($container->hasDefinition('texter'));
     }
 
+    public function testIfNotifierTransportsAreKnownByFrameworkExtension()
+    {
+        $container = $this->createContainerFromFile('notifier');
+
+        foreach ((new Finder())->in(\dirname(__DIR__, 4).'/Component/Notifier/Bridge')->directories()->depth(0)->exclude(['Mercure', 'SmsBiuras', 'MessageBird']) as $bridgeDirectory) {
```
Comment (jschaedl) [inline src/Symfony/Bundle/FrameworkBundle/Tests/DependencyInjection/FrameworkExtensionTest.php]: As soon as the repository `symfony/message-bird-notifier` and `symfony/sms-biuras-notifier` is available:

```suggestion
        foreach ((new Finder())->in(\dirname(__DIR__, 4).'/Component/Notifier/Bridge')->directories()->depth(0)->exclude('Mercure') as $bridgeDirectory) {
```

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @jschaedl.

**Pull request merged**